changelogs/3.1.0.md

```diff
@@ -0,0 +1,253 @@
+# 3.1.0
+
+Date: 2024-04-15
+Tag: 3.1.0
+
+## Overview
+
+
+3.1.0 is the 1st [stable][release_policy] version of the 3.1 release
+series. It introduces 31 improvements and resolves 35 bugs since
+3.0.1.
+
+The "stable" label means that we have all planned features implemented and we
+see no high-impact issues. However, if you encounter an issue, feel free to
+[report it][issues] on GitHub.
+
+[release_policy]: https://www.tarantool.io/en/doc/latest/dev_guide/release_management/#release-policy
+[issues]: https://github.com/tarantool/tarantool/issues
+
+
+## Compatibility
+
+Tarantool 3.1.x is backward compatible with Tarantool 2.11.x in the binary data
+layout, client-server protocol, and replication protocol.
+
+Please [upgrade][upgrade] using the `box.schema.upgrade()` procedure to unlock
+all the new features of the 3.1.x series.
+
+[upgrade]: https://www.tarantool.io/en/doc/latest/book/admin/upgrades/
+
+## Functionality added or changed
+
+### Core
+
+* Introduced fixed-size numeric field types: `uint8`, `int8`, `uint16`, `int16`,
+  `uint32`, `int32`, `uint64`, `int64`, `float32`, `float64` (gh-9548).
+* Now it is possible to set IPROTO request handler callbacks by using the new
+  Lua module `trigger`, for example:
+  ```lua
+  local trigger = require('trigger')
+  trigger.set('box.iproto.override.select', 'my_select', my_select_handler)
+  ```
+  The method works before the initial `box.cfg{}` call. Also, the method allows
+  setting multiple handlers for a single request type (gh-8138).
+* `box.iproto.override()` was switched to the universal trigger registry. As a
+  side effect, now this function does not raise an error when a wrong request
+  type is passed. All such errors are logged with `CRITICAL` level (gh-8138).
+
+### Raft
+
+* Now the leader resigns on the first encounter with the `ER_WAL_IO`
+  write error (gh-9399).
+
+### Lua
+
+* Now it is possible to add custom payload fields to `box.error` (gh-9104).
+* Added a `message` argument to the table constructor of `box.error.new`
+  (gh-9102).
+* Added a `prev` argument to the table constructor of `box.error.new` (gh-9103).
+* Now it is possible to access a payload field of an error's cause directly
+  from the error (gh-9106).
+* Increased the verbosity of `box.error`s serialization and added a new
+  `box_error_serialize_verbose` option to `compat` to retain old behaviour
+  (gh-9105).
+* Added autocompletion (including payload fields) to `box.error` (gh-9107).
+* Added the `level` argument to `box.error` and `box.error.new` to specify
+  the stack frame used for setting the error location (gh-9792).
+* Added key validation and comparison functions to the `key_def` module
+  (gh-9863).
+* Added a `tnt_election_leader_idle` metric to built-in `metrics`.
+
+#### Http client
+
+* Added the `http_version` option to the `request()`, `post()`,
+  `get()` and other request functions from the `http.client` module (gh-9806).
+
+### Box
```

Comment:
I guess we should move `box` bullets to the `core` section.

There is no `box` in [known sections](https://github.com/tarantool/tarantool/blob/da68227637776c34db5bf5db4dc33ecbbbe69777/tools/gen-release-notes#L14-L17), it is used by mistake.

Reply:
I believe @Totktonada will give you more details on the sources you mentioned. Still back [in time](https://github.com/tarantool/tarantool/releases) we had Box as a section for long time.

Reply:
I've looked over the items in these sections in the 3.1.0 release notes.

It looks like we should coalesce the sections, the component is actually the same.

Let's use Core or Box as you wish. Since Core is in the known section, I would prefer it. I don't remember exact reason why Core, not Box.

(Don't forget about the 'Bugs fixed' section.)

Reply:
It doesn't look done. Are not you forgot to push the changes?

Reply:
Looks like missed the update in the push. Done.

Reply:
Looks resolved now, thanks!